src/org/omegat/gui/glossary/GlossarySearcher.java

```diff
@@ -245,35 +255,70 @@ private static boolean tokenInTag(Token tok, List<Tag> tags) {
         return false;
     }
 
-    static void sortGlossaryEntries(List<GlossaryEntry> entries) {
+    /**
+     * sort glossary entries for test.
+     * 
+     * @param entries
+     */
+    void sortGlossaryEntries(List<GlossaryEntry> entries) {
+        final Collator srcLangCollator = Collator.getInstance(srcLang.getLocale());
+        final Collator targetLangCollator = Collator.getInstance(targetLang.getLocale());
+        sortGlossaryEntries(srcLangCollator, targetLangCollator, entries);
+    }
+
+    private void sortGlossaryEntries(Collator srcLangCollator, Collator targetLangCollator,
+            List<GlossaryEntry> entries) {
         entries.sort((o1, o2) -> {
             int p1 = o1.getPriority() ? 1 : 2;
             int p2 = o2.getPriority() ? 1 : 2;
             int c = p1 - p2;
             if (c == 0 && Preferences.isPreferenceDefault(Preferences.GLOSSARY_SORT_BY_SRC_LENGTH, true)
-                    && (o2.getSrcText().contains(o1.getSrcText())
-                            || o1.getSrcText().contains(o2.getSrcText()))) {
+                    && (o2.getSrcText().startsWith(o1.getSrcText())
+                            || o1.getSrcText().startsWith(o2.getSrcText()))) {
```

Comment:
In `org.omegat.gui.preferences.view.GlossaryPreference`, the default value of `Preferences.GLOSSARY_SORT_BY_SRC_LENGTH` is `false` but here is `true`.

It should be `false` for consistency.